lib/class-wp-rest-blocks-search-controller.php

```diff
@@ -0,0 +1,240 @@
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Install and activate a plugin
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function install_block( $request ) {
+		
+		include_once( ABSPATH . 'wp-admin/includes/file.php' );
+		include_once( ABSPATH . 'wp-admin/includes/plugin.php' );
+		include_once( ABSPATH . 'wp-admin/includes/class-wp-upgrader.php' );
+		include_once( ABSPATH . 'wp-admin/includes/plugin-install.php' );
+
+
+		$api = plugins_api(
+			'plugin_information',
+			array(
+				'slug'   => sanitize_key( wp_unslash( $request->get_param( 'slug' ) ) ),
+				'fields' => array(
+					'sections' => false,
+				),
+			)
+		);
+		if ( is_wp_error( $api ) ) {
+			$status['errorMessage'] = $api->get_error_message();
+			wp_send_json_error( $status );
+		}
+		$status['pluginName'] = $api->name;
+		$skin     = new WP_Ajax_Upgrader_Skin();
+		$upgrader = new Plugin_Upgrader( $skin );
+		$result   = $upgrader->install( $api->download_link );
+	
+		if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+			$status['debug'] = $skin->get_upgrade_messages();
+		}
+		if ( is_wp_error( $result ) ) {
+			$status['errorCode']    = $result->get_error_code();
+			$status['errorMessage'] = $result->get_error_message();
+			wp_send_json_error( $status );
+		} elseif ( is_wp_error( $skin->result ) ) {
+			$status['errorCode']    = $skin->result->get_error_code();
+			$status['errorMessage'] = $skin->result->get_error_message();
+			wp_send_json_error( $status );
+		} elseif ( $skin->get_errors()->has_errors() ) {
+			$status['errorMessage'] = $skin->get_error_messages();
+			wp_send_json_error( $status );
+		} elseif ( is_null( $result ) ) {
+			global $wp_filesystem;
+			$status['errorCode']    = 'unable_to_connect_to_filesystem';
+			$status['errorMessage'] = __( 'Unable to connect to the filesystem. Please confirm your credentials.' );
+			// Pass through the error from WP_Filesystem if one was raised.
+			if ( $wp_filesystem instanceof WP_Filesystem_Base && is_wp_error( $wp_filesystem->errors ) && $wp_filesystem->errors->has_errors() ) {
+				$status['errorMessage'] = esc_html( $wp_filesystem->errors->get_error_message() );
+			}
+			wp_send_json_error( $status );
+		}
+
+		$install_status = install_plugin_install_status( $api );
+
+		activate_plugin( $install_status['file'] );
+		
+		wp_send_json_success( $status );
+	}
+
+	/**
+	 * Search and retrieve blocks metadata
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+
+		if ( ! isset($_REQUEST[ 'search' ] ) ) {
+			return rest_ensure_response( array() );
+		}
+
+		$search_string = preg_quote( $_REQUEST[ 'search' ] );
```

Comment:
What is the purpose of `preg_quote` here?

Reply:
This was used when we had a regex to search for mock data. This is not required anymore and I have removed it.